lizmap/modules/lizmap/lib/Request/RemoteStorageRequest.php

```diff
@@ -0,0 +1,316 @@
+<?php
+/**
+ * Utilities for upload, delete and retreive resources from remote storage.
+ *
+ * @author
+ * @copyright 2012-2016 3liz
+ *
+ * @see      http://3liz.com
+ *
+ * @license Mozilla Public License : http://www.mozilla.org/MPL/
+ */
+
+namespace Lizmap\Request;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7;
+
+class RemoteStorageRequest
+{
+    protected static $webDAVNamespace = 'DAV:';
+
+    protected static $fileNameExpression = 'file_name(@selected_file_path)';
```

Comment:
This will need to be written in the documentation.
Side question,does `file_name( @selected_file_path   )` work ? or it needs to very explicit in the doc :)

Reply:
Agree, this should be documented. As you pointed out, the syntax of `file_name(@selected_file_path)` is specific and the feature doesn't work if there is trail spaces or other chars. Actually, `file_name(@selected_file_path)` is mandatory

Reply:
I just tried in QGIS desktop, using the UI to build an expression with click and click.

When double clicking items from the center to add the function on the left panel, it adds automatically two spaces ` file_name( ` before and after the inserted function, so I would expect a lot of users not being careful, even if it's clearly written in the documentation.

![image](https://github.com/3liz/lizmap-web-client/assets/1609292/62ff6428-c020-4e5a-8f97-67a589cbf963)

It would be way better to use a regepx. Tested very quickly `file_name\(\s*@selected_file_path\s*\)`

I guess trailing spaces works ? ` file_name(@selected_file_path)`

In the doc, it would still needed to say no multiline expression or comment.